xmodule/tests/test_item_bank.py

```diff
@@ -0,0 +1,498 @@
+        assert self.item_bank.validate()
+
+        # Set max_count to higher value than exists in library
+        self.item_bank.max_count = 50
+        result = self.item_bank.validate()
+        assert not result
+        self._assert_has_only_N_matching_problems(result, 4)
+        assert len(self.item_bank.selected_children()) == 4
+
+        # Add some capa problems so we can check problem type validation messages
+        self.item_bank.max_count = 1
+        assert self.item_bank.validate()
+        assert len(self.item_bank.selected_children()) == 1
+
+        # -1 selects all blocks from the library.
+        self.item_bank.max_count = -1
+        assert self.item_bank.validate()
+        assert len(self.item_bank.selected_children()) == len(self.item_bank.children)
+
+        assert "@@TODO make more assertions about the validation messages"
+
+    @patch(
+        'xmodule.modulestore.split_mongo.caching_descriptor_system.CachingDescriptorSystem.render',
+        VanillaRuntime.render,
+    )
+    @patch('xmodule.capa_block.ProblemBlock.author_view', dummy_render, create=True)
+    @patch('xmodule.x_module.DescriptorSystem.applicable_aside_types', lambda self, block: [])
+    def test_preview_view(self):
+        """ Test preview view rendering """
+        self._bind_course_block(self.item_bank)
+        rendered = self.item_bank.render(AUTHOR_VIEW, {'root_xblock': self.item_bank})
+        assert '<p>Hello world from problem 0</p>' in rendered.content
+        assert '<p>Hello world from problem 1</p>' in rendered.content
+        assert '<p>Hello world from problem 2</p>' in rendered.content
+        assert '<p>Hello world from problem 3</p>' in rendered.content
+
+    @patch(
+        'xmodule.modulestore.split_mongo.caching_descriptor_system.CachingDescriptorSystem.render',
+        VanillaRuntime.render,
+    )
+    @patch('xmodule.capa_block.ProblemBlock.author_view', dummy_render, create=True)
+    @patch('xmodule.x_module.DescriptorSystem.applicable_aside_types', lambda self, block: [])
+    def test_author_view(self):
+        """ Test author view rendering """
+        self._bind_course_block(self.item_bank)
+        rendered = self.item_bank.render(AUTHOR_VIEW, {})
+        assert '' == rendered.content
+        # content should be empty
+        assert 'LibraryContentAuthorView' == rendered.js_init_fn
+        # but some js initialization should happen
+
+    def _assert_has_only_N_matching_problems(self, result, n):
+        assert result.summary
+        assert StudioValidationMessage.WARNING == result.summary.type
+        assert f'only {n} have been selected' in result.summary.text
+
+
+@skip_unless_lms
+@ddt.ddt
+class TestItemBankForLms(ItemBankTestBase):
+    """
+    Test LMS ItemBank features: selection, analytics, resetting problems.
+    """
+
+    def _assert_event_was_published(self, event_type):
+        """
+        Check that a LegacyLibraryContentBlock analytics event was published by self.item_bank.
```

Comment:
Nit: `legacy` doesn't appear in the event name so this is slightly confusing to me

"Check that an analytics event (called edx.librarycontentblock.content.____ for backwards compatibility) was published" perhaps?

Reply:
Added a comment to library_content.py